### ISSUE TITLE
Script errors and goes into endless loop

### ISSUE THREAD

**Issue by blstrueb** [Francommit/win10_emulation_station#2]
HI Chris - looks like you have done some great work here, but this is where your script errors and goes into and endless loop of checking for the config file.  I deleted files from the .emulation station folder and restarted based on our troubleshooting, but same result.  Any ideas?

System ERROR:
The filename, directory name, or volume label syntax is incorrect.
& : The term 'C:\Users\Owner\.emulationstation\systems\retroarch\retroarch.exe' is not recognized as the name of a
cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again.
At line:159 char:3
+ & $retroarchExecutable
+   ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Owner\...h\retroarch.exe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Checking for config file...
Checking for config file...
Checking for config file...
Checking for config file...

Here is the log file:
lvl2: 	EmulationStation - v2.0.1a, built Mar  8 2015 - 16:10:39
lvl2: 	Creating surface...
lvl2: 	Created window successfully.
lvl2: 	Checking available OpenGL extensions...
lvl2: 	 ARB_texture_non_power_of_two: ok
lvl2: 	Loading system config file /etc/emulationstation/es_systems.cfg...
lvl0: 	es_systems.cfg file does not exist!
lvl0: 	Example config written!  Go read it at "C:/Users/Owner/.emulationstation/es_systems.cfg"!
lvl0: 	Error while parsing systems configuration file!

**Comment by Francommit** [comment]
Hey 	blstrueb,

Will try and replicate this when I get a chance. 
Bit busy with work at the moment!

Can I ask if you've tried running it more than once?

**Comment by blstrueb** [comment]
Thanks for the reply...
I re-ran it several times after deleting files in the folder as described.  I already have the latest mame update (192), and had already installed emulationstation when I ran the script - could that be the issue? Not ready to give up yet as this looks like a great UI.   Let me know if you think there is anything else I should be checking.  Thanks again

Brian

**Comment by Francommit** [comment]
Hey blstrueb!

Just ran the script on a fresh Windows 10 PC and it seems to have worked. 
I suspect it has something to do with your existing installation.

Have you tried removing the entire emulation-station folder?

"C:/Users/Owner/.emulationstation/"

In-addition. Given it's looping at the retro-arch step might be worth deleting the "RetroArch.7z" file from the resource folder in-case its corrupted. Low possibility but there's a chance that that's the case.

Give both of those a try and let me know how it goes.

Thanks!

**Comment by blstrueb** [comment]
I have not removed the entire folder, only the contents.  I will try your suggestions and let you know.  Do I need to have the rims in a certain directory, or does the script look for them on my machine?

Thanks again,

**Comment by Francommit** [comment]
I assume you mistyped roms? 

The script will download everything it needs to the 'requirements' folder each time it runs so you shouldn't have to worry about the files themselves.

**Comment by blstrueb** [comment]
Yes roms - my apologies.  Must take a while to download them all.  I will let u know...thanks again!

**Comment by blstrueb** [comment]
Deleted the folder, still hanging up.  It does not appear to be downloading the retroarch.exe file.  I can download separately - do I extract directly to the systems\retroarch folder or cores?

Thanks

**Comment by blstrueb** [comment]
Also just wondering if there are some regional issues - I am in the US...

**Comment by Francommit** [comment]
Can you post a screenshot of your downloaded folder 'requirements'.

**Comment by blstrueb** [comment]
It creates the folder, but it is empty.
![image](https://user-images.githubusercontent.com/34480175/34157603-f1d060fa-e487-11e7-91e2-a50b2e97f129.png)

**Comment by Francommit** [comment]
Ahh ok. Could I get you to re-download the zip file then extract everything in THAT folder.

C:\requirements

Then from there, go to the script in an Admin window of power shell and give it a run.

I suspect it may have been due to problems with extra Admin permissions being required when you're running scripts directly from the root of the C: drive.

Let me know how that goes!

**Comment by blstrueb** [comment]
To clarify, extract retroarch zip file to requirements folder?

**Comment by Francommit** [comment]
Sorry didn't explain properly. The entire repository:

![image](https://user-images.githubusercontent.com/11533365/34158569-0ae327c0-e51a-11e7-8c20-5bf5408b0c21.png)

**Comment by blstrueb** [comment]
Ok - I will give it a shot tonight. 

Thanks again

**Comment by blstrueb** [comment]
Same result...may have to call this one

![image](https://user-images.githubusercontent.com/34480175/34182329-5fdf2b5a-e4db-11e7-963c-c8a8b403282d.png)

**Comment by Francommit** [comment]
What's your email mate? I'll PM you there.

**Comment by blstrueb** [comment]
Brian@audiofusiondj.com

**Comment by blstrueb** [comment]
Working well now - thanks!

**Comment by stephenegriffin** [comment]
What was resolution on this? I've got the exact same issue on my machine. I've deleted the .emulationstation and requirements folders and still get the loop on "Checking for config file..."

**Edit:** Looking at the script, we're waiting for emulationstation.exe to create a config file. This reminds me that before I came here, I had tried to install ES myself. Maybe I should let the script install it instead? So I uninstall ES from Apps and Features and try the script again. Still fails. Apparently, choco thinks it's still installed, so it doesn't reinstall.

So I ran the choco install with --force. And now the script gets past the config bit!

Conclusion: Step 3 needs work. ES can be missing, or ES can be present, but for whatever reason does not generate a config file. Both conditions should be handled.
